content/client-lib-development-guide/features.textile

```diff
@@ -111,7 +111,9 @@ h3(#rest-auth). Auth
 ** @(RSA3b)@ For REST requests, the token string is Base64 encoded and used in the @Authorization: Bearer@ header
 ** @(RSA3c)@ For Realtime connections, the querystring param @accessToken@ is appended to the URL endpoint
 * @(RSA4)@ Token Auth is used if @useTokenAuth@ is set to true, or if @useTokenAuth@ is unspecified and any one of the following conditions are met: a @clientId@ is specified; @authUrl@ or @authCallback@ is provided; an explicit @token@ or @TokenDetails@ is provided
-** @(RSA4a)@ When a @token@ or @tokenDetails@ is used to instance the library, and no means to renew the token is provided (either an API key, @authCallback@ or @authUrl@), if the server responds with a token error (401 HTTP status code and an Ably error value @40140 <= code < 40150@), then the client library should indicate an error, not retry the request and in the case of the realtime library, transition the connection to the @FAILED@ state
+** @(RSA4a)@ When a @token@ or @tokenDetails@ is used to instance the library, and no means to renew the token is provided (either an API key, @authCallback@ or @authUrl@):
+*** @(RSA4a1)@ if the server responds with a token error (401 HTTP status code and an Ably error value @40140 <= code < 40150@), then the client library should indicate an error, not retry the request and in the case of the realtime library, transition the connection to the @FAILED@ state
+*** @(RSA4a2)@ a message at @info@ log level should be logged indicating that no means has been provided to renew the supplied token, with a link to a "solution article":https://help.ably.io/a_suitable_article with configuration advice relating to token authentication. 
```

Comment:
This is wrong IMHO for two reasons:

* `help.ably.io` follows a format for automatic redirects based on an Ably error code. All solutions are are `support.ably.io`, however I fear that coding that URL into the client libs is a big mistake.  At least with an error code and `help.ably.io/errors/[CODE]` we can guarantee that URL won't change, even if we changed support systems down the line.
* Why can't we instead simply state that all errors emitted with an error code automatically include a URL to `https://help.ably.io/errors/[CODE]` as a single new requirement and thus we won't need a requirement like this for every possible error in this spec.

Reply:
> help.ably.io follows a format for automatic redirects based on an Ably error code. All solutions are are support.ably.io, however I fear that coding that URL into the client libs is a big mistake. At least with an error code and help.ably.io/errors/[CODE] we can guarantee that URL won't change, even if we changed support systems down the line.

Right, that URL was clearly provisional. We can assign an error code for this.

> Why can't we instead simply state that all errors emitted with an error code automatically include a URL to https://help.ably.io/errors/[CODE] as a single new requirement and thus we won't need a requirement like this for every possible error in this spec.

For situations where we emit or return an error then we can, but this isn't an error.